test/shared/roundtrip.e2e.test.ts

```diff
@@ -0,0 +1,56 @@
+import { test, expect, Page } from "@playwright/test";
+import {
+    switchMode,
+    clearEditor,
+    editorSelector,
+    enterTextAsMarkdown,
+} from "../e2e-helpers";
+
+test.describe.serial("roundtrip tests", () => {
+    let page: Page;
+    test.beforeAll(async ({ browser }) => {
+        page = await browser.newPage();
+        await page.goto("/empty.html");
+        await switchMode(page, "markdown");
+    });
+
+    test.afterAll(async () => {
+        await page.close();
+    });
+
+    for (const [markdown] of [
+        //Basic commonmark
+        ["plain"],
+        ["*italic*"],
+        ["_italic_"],
+        ["**bold**"],
+        ["__bold__"],
+        ["# H1"],
+        ["## H2"],
+        ["[link](http://www.example.com)"],
+        ["![Image](http://www.example.com/pretty.png)"],
+        ["> blockquote"],
+        ["* List Item"],
+        ["- List Item"],
+        ["1. List Item"],
+        ["2) List Item"],
+        ["lol\n\n---\n\nlmao"],
+        ["lol\n\n***\n\nlmao"],
+        ["`code`"],
+        //TODO: Codeblock does weird things roundtripping: Adds an extra space
+        //['```javascript\ncodeblock\n```' ],
+
+        //Escape character
+        [String.raw`\# not a header`],
+        [String.raw`- \# list item (not header)`],
+    ] as const) {
+        test(`should make markdown -> richtext -> markdown round trip '${markdown}'`, async () => {
```

Comment:
This test title is making the test output less readable because it honours the `\n` special char. A way to solve that is to do something like this:
```
`should make markdown -> richtext -> markdown round trip ${JSON.stringify(markdown)}`
```
![Screenshot 2025-02-04 at 10 15 34](https://github.com/user-attachments/assets/39e65fe2-1a1a-4545-ae40-4762d65ca724)